wallet/remote.go

```diff
@@ -0,0 +1,120 @@
+package wallet
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+// Remote is an interface, which instances are used to communicate with the perun-cardano-wallet server
+type Remote interface {
+	// CallSign is the endpoint for signing data with the perun-cardano-wallet
+	CallSign(SigningRequest) (SigningResponse, error)
+	// CallVerify is the endpoint for verifying signatures with the perun-cardano-wallet
+	CallVerify(VerificationRequest) (VerificationResponse, error)
+	// CallKeyAvailable is the endpoint for verifying that the connected perun-cardano-wallet has the private key to
+	// a given PubKey
```

Comment:
It is a bit weird now, but I think renaming it to Address is still correct.
We will have to change the types in `encoding.go` once `Address` is no longer secretly just a public key, but that feels like the right thing to do, especially since I introduced smart constructors for them as you suggested.

This means currently an `Address` is actually equivalent to a `PubKey` in the cardano codebase on a json-level. The (Signing|Verification)(Request|Response) structs abuse this to avoid adding yet another `PubKey` type just for json encoding, but we can easily change this once we arrive at the final version of `Address` without changing the signature of the smart constructors / the decoding helpers (yayy)!
Saaaaadly, I cannot *not export* the struct members because they need to have json tags... I'm starting not to like go....